bin/proving-service/src/utils.rs

```diff
@@ -148,27 +148,24 @@ pub async fn create_response_with_error_message(
 /// Checks if a port is available for use.
 ///
 /// # Arguments
-/// * `host` - The host to bind to.
 /// * `port` - The port to check.
-/// * `port_name` - A descriptive name for the port (for logging purposes).
+/// * `service` - A descriptive name for the service (for logging purposes).
 ///
 /// # Returns
 /// * `Ok(())` if the port is available.
 /// * `Err(ProvingServiceError::PortAlreadyInUse)` if the port is already in use.
-pub fn check_port_availability(
-    host: &str,
-    port: u16,
-    port_name: &str,
-) -> Result<(), ProvingServiceError> {
-    let addr = format!("{}:{}", host, port);
+pub fn check_port_availability(port: u16, service: &str) -> Result<(), ProvingServiceError> {
+    let addr = format!("{}:{}", PROXY_HOST, port);
     match TcpListener::bind(&addr) {
         Ok(_) => {
             // Port is available, we can proceed
-            tracing::info!("Port {} is available for {}", port, port_name);
+            tracing::info!("Port {} is available for {}", port, service);
             Ok(())
         },
         Err(e) => {
             // Port is already in use, log an error and return an error
+            let error_msg = format!("{} port {} is already in use: {}", service, port, e);
+            tracing::error!("{}", error_msg);
             Err(ProvingServiceError::PortAlreadyInUse(e, port))
```

Comment:
Does it make sense to both log and return the error here? Basically, if the error will be logged somewhere downstream, this logging may be redundant.

Reply:
I tend to agree with this, I also wouldn't log the happy path on `info` but rather `debug` if at all.

As an aside I tend to like the `inspect` style methods for this sort of thing to make clear nothing "smart" is happening, and one should prefer logged fields over print fmts imo:

```rust
TcpListener::bind(&add)
    .inspect_ok(|_| tracing::info!(%service, %port, "Port is available"))
    .inspect_err(|err| tracing::error!(%service, %port, %err, "Port unavailable"))
    .map_err(|err| ProvingServiceError::PortAlreadyInUse(..., port))
```

Reply:
Thanks! I'm changing to this structure. Also, removed the `error` log and replaced the info with debug.